### PR TITLE
Put back build steps into build-rpm

### PR DESCRIPTION
**Pull request by clnperez**
**\- What I did**

**\- How I did it**

**\- How to verify it**

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

Since you have to have the source for the src rpm.

Signed-off-by: Christy Perez christy@linux.vnet.ibm.com

### CODE REVIEW

**Comment by clnperez** [comment]
Similar to @epiloque, I was adding a builder for ppc64le & fedora but ran into the same issue. I was working on fixing it, and have since seen PR #27844, which was my first thought. The bigger problem is that you still need the source later, so I tried just pulling the zip files of the commits but ran into a weird build error: "No build ID note found," so it seems like the only thing to do is the git clones again. :(

File: hack/make/build-rpm [hunk 1]
```diff
@@ -92,10 +92,38 @@ set -e
 			RUN mkdir -p /go/src/github.com/docker && mkdir -p /go/src/github.com/opencontainers
 		EOF
 
+		echo "ENV RUNC_COMMIT $(grep "^RUNC_COMMIT" hack/dockerfile/install-binaries.sh | cut -d"=" -f 2)" >> "$DEST/$version/Dockerfile.build"
+		echo "ENV CONTAINERD_COMMIT $(grep "^CONTAINERD_COMMIT" hack/dockerfile/install-binaries.sh | cut -d"=" -f 2)" >> "$DEST/$version/Dockerfile.build"
+        echo "ENV LIBNETWORK_COMMIT $(grep "^LIBNETWORK_COMMIT" hack/dockerfile/install-binaries.sh | cut -d"=" -f 2)" >> "$DEST/$version/Dockerfile.build"
```
Comment (tophj-ibm) [inline hack/make/build-rpm]: spaces v tabs 😄

**Comment by clnperez** [comment]
But there's an error during the rpm %build step:
```

---> Making bundle: dynbinary (in bundles/1.13.0-dev/dynbinary)
Building: bundles/1.13.0-dev/dynbinary-client/docker-1.13.0-dev
Created binary: bundles/1.13.0-dev/dynbinary-client/docker-1.13.0-dev
Building: bundles/1.13.0-dev/dynbinary-daemon/dockerd-1.13.0-dev
# github.com/docker/docker/daemon
.gopath/src/github.com/docker/docker/daemon/daemon_unix.go:715: cannot use nw (type []*net.IPNet) as type *net.IPNet in argument to "github.com/docker/libnetwork/types".GetIPNetCanonical
.gopath/src/github.com/docker/docker/daemon/daemon_unix.go:716: nw.IP undefined (type []*net.IPNet has no field or method IP)
.gopath/src/github.com/docker/docker/daemon/daemon_unix.go:716: nw.Mask undefined (type []*net.IPNet has no field or method Mask)
.gopath/src/github.com/docker/docker/daemon/daemon_unix.go:718: nw.IP undefined (type []*net.IPNet has no field or method IP)
```

**Comment by LK4D4** [comment]
@clnperez I'd really like to avoid git clones :/ It's insanely hard to maintain.

**Comment by LK4D4** [comment]
Maybe someone knows what is build ID note? ping @tiborvass @mlaventure

**Comment by LK4D4** [comment]
It's something with way docker-proxy installed, I'm not sure if `git clones` help, did it work for you @clnperez ?

**Comment by LK4D4** [comment]
I've noticed that we also do not put sources of grimes to SOURCES. build note ID is still mystery for me, I've tried many things already :)

**Comment by tiborvass** [comment]
We're onto something. I added questions in justin's commit: https://github.com/docker/docker/commit/3996975b0840033bda3919440d122d734a4bc66b

**Comment by clnperez** [comment]
@tiborvass Yay for progress! There's still need for the source later on in the build though. So should I redo my PR to do the curl of the zips? I still have that in an old branch.

**Comment by clnperez** [comment]
Closing this one since #27844 was reworked to fix it. :tada:

**Pull request closed without merging**